src/Reflection/ClassReflection.php

```diff
@@ -769,35 +772,53 @@ public function getTypeAliases(): array
 			$typeAliasImportTags = $resolvedPhpDoc !== null ? $resolvedPhpDoc->getTypeAliasImportTags() : [];
 			$typeAliasTags = $resolvedPhpDoc !== null ? $resolvedPhpDoc->getTypeAliasTags() : [];
 
-			$importedAliases = array_map(function (TypeAliasImportTag $typeAliasImportTag): TypeAlias {
+			// prevent circular imports
+			if (array_key_exists($this->getName(), self::$resolvingTypeAliasImports)) {
+				throw new \PHPStan\Type\CircularTypeAliasDefinitionException();
```

Comment:
Everyone who's calling getTypeAliases() should be prepared to catch this. Wouldn't it be better to just return an empty array? Every situation like that should be caught by the rule anyway.

This is similar to FileTypeMapper that also silently returns empty ResolvedPhpDocBlock.

Reply:
Under normal circumstances, you should never get this exception from a direct call to getTypeAliases(). It's only thrown if the method is called in recursion a [couple of lines below](https://github.com/phpstan/phpstan-src/pull/460/files/7f066e5fb9ccc7f4106829e775f45ecde86c649d..3519a52cc99046deec8a721a39ccdc0ddd167ebb#diff-1cb8c0f16a3848b23fbd09ac0ce1f5ef06d9e8365d2aded0c6028b8035c78c6fR797-R801), at which point it is caught and transformed into an invalid `TypeAlias`.

Returning `[]` caused some of the circular aliases to be left unresolved, and thus undetectable by the rule without having to reimplement the import logic there.